src/lib/mdns/Resolver.h

```diff
@@ -44,9 +44,12 @@ struct ResolvedNodeData
     }
 
     PeerId mPeerId;
-    Inet::InterfaceId mInterfaceId;
-    Inet::IPAddress mAddress;
-    uint16_t mPort;
+    Inet::IPAddress mAddress         = Inet::IPAddress::Any;
+    Inet::InterfaceId mInterfaceId   = INET_NULL_INTERFACEID;
+    uint16_t mPort                   = 0;
+    bool mSupportsTcp                = false;
```

Comment:
We also have this nonsensical inference for TCP support from `_matterX._tcp` and `_udp`:

```
MdnsServiceProtocol GetProtocolInType(const char * type)
{
    const char * deliminator = strrchr(type, '.');

    if (deliminator == NULL)
    {
        ChipLogError(Discovery, "Failed to find protocol in type: %s", type);
        return MdnsServiceProtocol::kMdnsProtocolUnknown;
    }

    if (strcmp("._tcp", deliminator) == 0)
    {
        return MdnsServiceProtocol::kMdnsProtocolTcp;
    }
    if (strcmp("._udp", deliminator) == 0)
    {
        return MdnsServiceProtocol::kMdnsProtocolUdp;
    }

    ChipLogError(Discovery, "Unknown protocol in type: %s", type);
    return MdnsServiceProtocol::kMdnsProtocolUnknown;
}
```

Since this PR is adding support for correctly inferring TCP support, can we also remove this code that does that incorrectly?

Reply:
I think the code is pasted is not for inferring TCP support, but simply for extracting the protocol from a DNS service type so it can be used to distinguish Matter operational nodes (using X._tcp) from commissionable/commissioner ndes (using X._udp). The result of this function is not used at the moment, but it doesn't seem to be related to the "T" TXT entry.

Reply:
This by the way is tracked by https://github.com/project-chip/connectedhomeip/issues/8214.

I know it's our usual preference to have granular PRs, and so if we don't want to do this cleanup now, I suppose that is OK.  But I wanted to raise the issue that users of the feature in this PR may be confused by the preexisting code in the sdk which implies the same functionality (ability to deduce whether discovered nodes support TCP), but does so completely incorrectly.

Reply:
@Damian-Nordic ,

Ah I see.  Yes, I guess it makes sense to preserve the service name hierarchy and this field is the protocol field.

OK, will close #8214.